lib/Server.js

```diff
@@ -630,7 +635,21 @@ class Server {
       options.https.key = options.https.key || fakeCert;
       options.https.cert = options.https.cert || fakeCert;
 
-      if (!options.https.spdy) {
+      // options.https.spdy is here for slightly better backwards compatability,
+      // since a user will probably provide this deprecated option if they
+      // are expecting the spdy server to be used.
+      const isHttp2 = options.http2 || options.https.spdy;
```

Comment:
What do you think should we enable `http2` be default when user use `https: true` (without `http2` option) for better DX?

Reply:
I would prefer this since h2 is the standard for most CDNs going forward (or should be). Would be good to mimic the greater movement across the web. I originally opened the issue because I wrongly assumed I was getting h2 by default.

Reply:
I changed it so that h2 is only prevented if http2 is explicitly set to false.